process/interceptors/processor/hdrInterceptorProcessor.go

```diff
@@ -81,8 +81,8 @@ func (hip *HdrInterceptorProcessor) Save(data process.InterceptedData, _ core.Pe
 	hip.headers.AddHeader(interceptedHdr.Hash(), interceptedHdr.HeaderHandler())
 
 	if common.ShouldBlockHavePrevProof(interceptedHdr.HeaderHandler(), hip.enableEpochsHandler, common.EquivalentMessagesFlag) {
-		ok = hip.proofs.AddProof(interceptedHdr.HeaderHandler().GetPreviousProof())
-		log.Trace("HdrInterceptorProcessor.AddProof: add previous proof", "intercepted header hash", interceptedHdr.Hash(), "added", ok)
+		ok = hip.proofs.UpsertProof(interceptedHdr.HeaderHandler().GetPreviousProof())
```

Comment:
I think we can remove this one, and update to upsert the one in VerifyHeaderWithProof

also it could be removed also from saveProofForPreviousHeaderIfNeeded as it should have already added on verifyHeaderWithProof

Reply:
removed duplicated add prev proofs